awscli/customizations/cloudformation/deployer.py

```diff
@@ -178,10 +179,13 @@ def wait_for_execute(self, stack_name, changeset_type):
 
         # Poll every 5 seconds. Optimizing for the case when the stack has only
         # minimal changes, such the Code for Lambda Function
-        waiter.config.delay = 5
+        waiter_config = {
+            'Delay': 5,
+            'MaxAttempts': 720,
```

Comment:
1 hour seems like a long time

Reply:
Yeah the `720` is a bit concerning, given a typical waiter is an order of magnitude smaller (i.e. `30` or `45`). However, it is not too bad given the [pricing model](https://aws.amazon.com/cloudformation/pricing/) of CloudFormation where it should not cost money to be making these extra calls. Also there is not really any documentation on request rate limits, so we should be fine there as well. Out of curiosity, what was the previous total time waited (i.e. delay * max_attempts)?

Reply:
All three waiters used in this file (`change_set_create_complete`, `stack_create_complete`, and `stack_update_complete`) all have a default delay of 30 seconds with a max attempts of 120. 

I picked 720 to compensate for reducing the delay so that it will still wait 1 hour. 

The original commit of this command used the default waiter behavior. 
https://github.com/aws/aws-cli/blob/991e09eeb185116ac9015c3e8c6382cbbc1abae0/awscli/customizations/cloudformation/deployer.py#L159-L168
This was later reduced to 5 seconds, thus reducing the total time to wait by a factor of 6.

The `change_set_create_complete` waiter started at a delay of 10 and the default max attempts of 120. All waiters were changed to a delay of 5 in the same commit. We haven't had any issue reports on this waiter as far as I know, so I didn't increase the max attempts there. Though the total time to wait here was cut in half.

Reply:
Alright given that explanation and since the `max_attempts` was already set at `120`, I am less concerned about increasing this.